lib/apps/timers.js

```diff
@@ -50,23 +50,32 @@ class BaseTimer {
 }
 
 class Timer extends BaseTimer {
-    constructor(base, interval) {
+    constructor(base, interval, frequency) {
         super();
 
         this._base = base;
         this._interval = interval;
+        this._frequency = frequency;
     }
 
     toString() {
-        return `[Timer ${this._base}, ${this._interval}]`;
+        return `[Timer ${this._base}, ${this._interval}, ${this._frequency}]`;
     }
 
     _nextTimeout() {
+        let interval = this._interval;
+        if (this._frequency === 0) { // End timer because it will never execute
+            this.end();
+            return 86400000;
+        }
+        if (this._frequency !== null) {
+            interval /= this._frequency;
```

Comment:
This will produce a fractional value if interval and frequency are not divisible. I am not sure which way we want to round to be honest.
Also, implementing "3 times a day" as "exactly every 8 hours" might not be the best.

Reply:
Good point -- we don't want to send notifications to the user while they are sleeping after all. One possible solution we could do is to create context values for waking and sleeping times and clamp frequency intervals between those times. This solution is not without its problems but should solve most issues, and at least the most common issues, encountered.

Once Almond supports multi-stage commands -- once it can modify previous commands based on user feedback -- we can implement that for timers. We can then present the user with a list of times and let the user edit the list. Until then, however, I think this solution should cover most cases.

Reply:
As for the rounding, I'm not sure if there's any benefit to always rounding up vs always rounding down in this scenario. I think we should do a mathematical rounding instead.

Reply:
Well, the best solution for now, and the one that Monica proposed, is to recognize that the interval is one day or one week (which are the most common case) and hardcode reasonable times. For example, if the interval is one week, you need to make sure the timer fires always at the same time of day.